docker/catalog/Dockerfile

```diff
@@ -0,0 +1,31 @@
+# Official lightweight Node.js image
+# https://hub.docker.com/_/node
+FROM node:18.10-slim
+
+# Arbitrary but conventional working directory
+WORKDIR /usr/src/app
+
+# Install NPM dependencies
+COPY package.json package-lock.json ./
+COPY packages/custom-elements-manifest-tools/package.json packages/custom-elements-manifest-tools/
+COPY packages/catalog-api/package.json packages/catalog-api/
+COPY packages/catalog-server/package.json packages/catalog-server/
+RUN npm ci
+
+# Build the server
+COPY tsconfig.base.json ./
+COPY packages/custom-elements-manifest-tools/ packages/custom-elements-manifest-tools/
```

Comment:
why are these COPYs separate from the package.json COPYs above?

Reply:
The first ones are just the `package.json` files, so that `npm ci` works. The second ones are the actual code, so that `npm run build` works.

The reason these are separate blocks is for efficient Docker image layering. It means that if the `package.json` files didn't change, then the `npm ci` result can be restored from cache.